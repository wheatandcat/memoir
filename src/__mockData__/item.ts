import { ItemQuery } from 'queries/api/index';
type Item = NonNullable<ItemQuery['item']>;

export const item = (option?: Partial<Item>): Item => ({
  id: '1',
  title: '買い物',
  categoryID: 9,
  date: '2021-01-01T00:00:00+09:00',
  like: false,
  dislike: false,
  createdAt: '2021-01-01T00:00:00+09:00',
  updatedAt: '2021-01-01T00:00:00+09:00',
  ...option,
});

export const items = (): Item[] => [
  item(),
  item({
    id: '2',
    title: '本を読む',
    categoryID: 2,
    date: '',
    dislike: false,
    like: false,
  }),
];
