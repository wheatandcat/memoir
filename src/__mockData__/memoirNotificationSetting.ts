import { State as MemoirNotificationSettingState } from 'hooks/useMemoirNotificationSetting';

export const memoirNotificationSetting = (
  option?: Partial<MemoirNotificationSettingState>
): MemoirNotificationSettingState => ({
  dayOfWeek: 0,
  hours: 0,
  minutes: 0,
  notification: false,
  ...option,
});
