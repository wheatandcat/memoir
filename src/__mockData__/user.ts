import { User } from 'queries/api/index';

export const user = (option?: Partial<User>): User => ({
  id: '1',
  displayName: 'test',
  image: '',
  updatedAt: '2021-01-01T00:00:00+09:00',
  createdAt: '2021-01-01T00:00:00+09:00',
  ...option,
});
