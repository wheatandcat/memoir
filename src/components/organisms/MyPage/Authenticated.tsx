import React, { memo } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';
import theme from 'config/theme';
import View from 'components/atoms/View';
import Text from 'components/atoms/Text';
import { User } from 'store/atoms';
import UserImage from 'components/molecules/User/Image';

export type Props = {
  user: User;
  onUpdateProfile: () => void;
  onLogout: () => void;
};

const Authenticated: React.FC<Props> = (props) => {
  return (
    <View style={styles.root}>
      <View mt={4}>
        <UserImage image={props.user.image} />
      </View>
      <View m={4}>
        <TouchableOpacity onPress={props.onUpdateProfile}>
          <View style={styles.userInfo}>
            <Text>{props.user?.displayName || '表示名の設定がありません'}</Text>
            <MaterialIcons
              name="chevron-right"
              size={23}
              color={theme().color.secondary.main}
            />
          </View>
        </TouchableOpacity>
      </View>
      <View mt={5}>
        <TouchableOpacity onPress={props.onLogout}>
          <View style={styles.logout}>
            <Text>ログアウト</Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    backgroundColor: theme().color.background.main,
    height: '100%',
    alignItems: 'center',
  },
  userInfo: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  logout: {
    backgroundColor: theme().color.primary.main,
    width: 200,
    paddingVertical: theme().space(3),
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default memo(Authenticated);
