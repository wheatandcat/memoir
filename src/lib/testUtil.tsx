import React from 'react';
import { GraphQLHandler, GraphQLRequest } from 'msw';
import {
  ApolloClient,
  ApolloProvider,
  createHttpLink,
  InMemoryCache,
} from '@apollo/client';
import { render } from '@testing-library/react-native';
import { server } from 'mocks/server';
import fetch from 'cross-fetch';

const link = createHttpLink({
  uri: 'http://localhost:8080/query',
  fetch,
  credentials: 'same-origin',
});

const client = new ApolloClient({
  link,
  uri: 'http://localhost:8080/query',
  cache: new InMemoryCache(),
});

export const testRenderer =
  (children: React.ReactNode) =>
  (responseOverride?: GraphQLHandler<GraphQLRequest<never>>) => {
    if (responseOverride) {
      server.use(responseOverride);
    }
    return render(<ApolloProvider client={client}>{children}</ApolloProvider>);
  };
