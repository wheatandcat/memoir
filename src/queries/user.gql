query User {
  user {
    id
    displayName
  }
}
