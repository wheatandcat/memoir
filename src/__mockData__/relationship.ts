import type { RelationshipsQuery } from "queries/api/index";
import { user } from "./user";

export type Relationship = NonNullable<
  EdgesNode<RelationshipsQuery["relationships"]>
>;

export const relationship = (option?: Partial<Relationship>): Relationship => ({
  id: "1",
  followedId: "test",
  followerId: "test",
  user: user(),
  createdAt: "2021-01-01T00:00:00+09:00",
  updatedAt: "2021-01-01T00:00:00+09:00",
  ...option,
});

export const relationships = (): Relationship[] => [
  relationship(),
  relationship({
    id: "2",
    user: user(),
  }),
  relationship({
    id: "3",
    user: user(),
  }),
  relationship({
    id: "4",
    user: user(),
  }),
];
