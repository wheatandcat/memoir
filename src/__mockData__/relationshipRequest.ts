import { RelationshipRequestsQuery } from 'queries/api/index';
import { user } from './user';

export type RelationshipRequest = NonNullable<
  EdgesNode<RelationshipRequestsQuery['relationshipRequests']>
>;

export const relationshipRequest = (
  option?: Partial<RelationshipRequest>
): RelationshipRequest => ({
  id: '1',
  followedId: 'test',
  followerId: 'test',
  user: user(),
  createdAt: '2021-01-01T00:00:00+09:00',
  updatedAt: '2021-01-01T00:00:00+09:00',
  ...option,
});

export const relationshipRequests = (): RelationshipRequest[] => [
  relationshipRequest(),
  relationshipRequest({
    id: '2',
    user: user(),
  }),
  relationshipRequest({
    id: '3',
    user: user(),
  }),
  relationshipRequest({
    id: '4',
    user: user(),
  }),
];
