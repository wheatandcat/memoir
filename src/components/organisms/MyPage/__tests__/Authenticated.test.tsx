import React from 'react';
import { testRenderer } from 'lib/testUtil';
import { screen } from '@testing-library/react-native';
import { user } from '__mockData__/user';
import { relationships } from '__mockData__/relationship';
import Authenticated, { Props } from '../Authenticated';

const propsData = (): Props => ({
  user: {
    ...user(),
    userID: '',
  },
  relationships: relationships(),
  relationshipRequestCount: 3,
  onUpdateProfile: jest.fn(),
  onLogout: jest.fn(),
  onAddShareUser: jest.fn(),
  onRelationshipRequests: jest.fn(),
  deleting: false,
  onDeleteRelationship: jest.fn(),
});

describe('components/organisms/MyPage/Authenticated.tsx', () => {
  it('正常にrenderすること', () => {
    testRenderer(<Authenticated {...propsData()} />)();
    expect(screen.findAllByText('')).toBeTruthy();
  });
});
