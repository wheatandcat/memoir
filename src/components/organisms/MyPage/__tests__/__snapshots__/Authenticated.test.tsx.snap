// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/MyPage/Authenticated.tsx 正常にrenderすること 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F2F2F2",
      "height": "100%",
    }
  }
>
  <View
    mt={4}
  >
    <Memo(UserImage)
      image=""
    />
  </View>
  <View
    m={4}
  >
    <ForwardRef
      onPress={[MockFunction]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          fontFamily="RobotoCondensed-Bold"
          variants="body"
        >
          test
        </Text>
        <Icon
          allowFontScaling={false}
          color="#362740"
          name="chevron-right"
          size={23}
        />
      </View>
    </ForwardRef>
  </View>
  <View
    mt={5}
  >
    <ForwardRef
      onPress={[MockFunction]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E3C95D",
            "borderRadius": 25,
            "justifyContent": "center",
            "paddingVertical": 16,
            "width": 200,
          }
        }
      >
        <Text
          fontFamily="RobotoCondensed-Bold"
          variants="body"
        >
          ログアウト
        </Text>
      </View>
    </ForwardRef>
  </View>
</View>
`;
